WrightTools/artists/_interact.py

```diff
@@ -0,0 +1,381 @@
+            this_index = int(sliders[axis.natural_name].val)
+            sliders[axis.natural_name].valtext.set_text(ticklabels[this_index])
+            if verbose:
+                print(axis.natural_name, sliders[axis.natural_name].val, ticklabels[this_index])
+            slices.append(slice(this_index, this_index + 1))
+        else:
+            slices.append(slice(None))
+    return slices
+
+
+def gen_ticklabels(points):
+    step = np.diff(points).min()
+    if step == 0:
+        ticklabels = ['NaN' for point in points]
+        return ticklabels
+    ordinal = np.log10(np.abs(step))
+    ndigits = -int(np.floor(ordinal))
+    if ndigits < 0:
+        ndigits += 1
+        fmt = "{0:0.0f}"
+    else:
+        fmt = "{" + "0:.{0}f".format(ndigits) + "}"
+    ticklabels = [fmt.format(round(point, ndigits)) for point in points]
+    return ticklabels
+
+
+def norm(arr, signed, ignore_zero=True):
+    if signed:
+        norm = np.nanmax(np.abs(arr))
+    else:
+        norm = np.nanmax(arr)
+    if norm != 0 and ignore_zero:
+        arr /= norm
+    return arr
+
+
+def set_aspect(xaxis, yaxis):
+    if xaxis.units == yaxis.units:
+        xr = xaxis.max() - xaxis.min()
+        yr = yaxis.max() - yaxis.min()
+        aspect = np.abs(yr / xr)
+        if 3 < aspect or aspect < 1 / 3.:
+            raise Warning(
+                "units agree, but aspect {0} required for equal spacing is too"
+                / + "narrow.".format(aspect)
+            )
+            aspect = np.clip(aspect, 1 / 3., 3.)
+            print("using aspect {0} instead".format(aspect))
+    else:
+        aspect = 1
+    return aspect
+
+
+def interact2D(data, xaxis=0, yaxis=1, channel=0, local=False, verbose=True):
+    """ Interactive 2D plot of the dataset.
+    Side plots show x and y projections of the slice (shaded gray).
+    Left clicks on the main axes draw 1D slices on side plots at the coordinates selected.
+    Right clicks remove the 1D slices.
+    For 3+ dimensional data, sliders below the main axes are used to change which slice is viewed.
+
+    Parameters
+    ----------
+    data : WrightTools.Data object
+        Data to plot.
+    axis : string, integer, or data.Axis object (optional)
+        Expression or index of axis. Default is 0.
+    channel : string, integer, or data.Channel object (optional)
+        Name or index of channel to plot. Default is 0.
+    local : boolean (optional)
+        Toggle plotting locally. Default is False.
+    verbose : boolean (optional)
+        Toggle talkback. Default is True.
+    """
+    # unpack
+    channel = get_channel(data, channel)
+    xaxis, yaxis = get_axes(data, [xaxis, yaxis])
+    cmap = get_colormap(channel)
+    current_state = Bunch()
+    # create figure
+    nsliders = data.ndim - 2
+    if nsliders < 0:
+        print("note enough dimensions")
+        return
+    # TODO: implement aspect; doesn't work currently because of our incorporation of colorbar
```

Comment:
Resolved?

Reply:
no

Reply:
The TODO can stay as a comment, but should have an associated issue (I just don't want to loose track of things we intend to do)

Reply:
I would consider adding a date/initials to the TODO, just so that thought is attributable (git serves this purpose as well, but visibility is useful/the line could be edited for something like whitespace or refactoring which would obfuscate the git history)